client/blocks/shortcode/index.js

```diff
@@ -0,0 +1,56 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React, { Fragment } from 'react';
+import { omit } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import QueryShortcode from 'components/data/query-shortcode';
+import { getShortcode } from 'state/shortcodes/selectors';
+
+/**
+ * Local dependencies
+ */
+import ShortcodeFrame from './frame';
+
+const Shortcode = props => {
+	const { siteId, className, children, filterRenderResult, shortcode } = props;
+	const classes = classNames( 'shortcode', className );
+	let filteredShortcode = {};
+	if ( shortcode ) {
+		shortcode.body = shortcode.result;
+		filteredShortcode = filterRenderResult( omit( shortcode, 'shortcode' ) );
+	}
+	return (
+		<Fragment>
+			<QueryShortcode siteId={ siteId } shortcode={ children } />
+			<ShortcodeFrame
+				{ ...omit( props, 'siteId', 'filterRenderResult', 'shortcode', 'dispatch' ) }
+				{ ...filteredShortcode }
+				className={ classes }
+			/>
+		</Fragment>
+	);
+};
+
+Shortcode.propTypes = {
+	siteId: PropTypes.number.isRequired,
+	children: PropTypes.string.isRequired,
+	filterRenderResult: PropTypes.func,
+	className: PropTypes.string,
+	shortcode: PropTypes.string,
```

Comment:
I'm getting a warning for the `shortcode` prop:
> Warning: Failed prop type: Invalid prop `shortcode` of type `object` supplied to `Shortcode`, expected `string`.

The value returned by the `getShortcode` selector is an object.

Reply:
bah.

Reply:
fixed